### ISSUE TITLE
RuleFilter doesn't work for ....Non-Rules

### ISSUE THREAD

**Issue by seekM** [dictation-toolbox/Caster#540]
words.txt:

```
<<<ANY>>>
shift right click -> test
```

for [the spec](https://github.com/dictation-toolbox/Caster/blob/develop/castervoice/lib/ccr/core/nav.py#L56) in NavigationNon doesn't have an effect. 

Can someone confirm?

**Comment by seekM** [comment]
I fiddled with the source code a bit and added
 ```
            mp = MergePair(
                MergeInf.BOOT_NO_MERGE, MergeInf.GLOBAL, None, rule, False
            )  # copies not made at boot time, allows user to make permanent changes  
            self._run_filters(mp)    
```
between [these lines](https://github.com/dictation-toolbox/Caster/blob/develop/castervoice/lib/dfplus/merge/ccrmerger.py#L372-L373) in CCRMerger. Haven't tested it thoroughly but it seems to work. I don't know how CCRMerger works though and what implications this might have.

I'm a bit confused because I think it used to work in the past.

**Comment by LexiconCode** [comment]
> 
> 
> words.txt:
> 
> ```
> <<<ANY>>>
> shift right click -> test
> ```

try  `click -> test` I don't think that 'words.txt' works with multiple strings 'shift right click'

**Comment by seekM** [comment]
Doesn't work. AFAIK words.txt works for multiple strings (substrings). I use it to successfully replace multiple strings, e.g.

```
[is] greater than -> (greater than | größer als)
```

**Comment by LexiconCode** [comment]
> non: Suppose you want a non-CCR rule activated alongside your CCR rule. For instance, let's say there are some Python commands you want available while you're coding in Python, but which you know you'll never use as part of a command sequence and which you don't want in the CCR command set for performance / phonetic distinctness reasons. You can define non as the class of that rule and it will be instantiated and activated alongside the MergeRule you define it on.

I wonder if `non` is being treated like `MappingRule` which wouldn't be affected by filter rules.

**Comment by LexiconCode** [comment]
resolved in 1.0.0
